CuratedContent/CMakeTutorialAndReferenceResources.md

```diff
@@ -0,0 +1,87 @@
+The author is not currently a Kitware employee but is a significant contributor to CMake and collaborates closely with Kitware staff.
+The author starts evaluating and updating the book at the beginning of each new CMake release cycle (i.e. right after the first release candidate `vX.Y.0-rc1` is tagged).
+Through this process, the author often provides feedback to Kitware and help addressing issues with new CMake features and changes to existing CMake functionality which feeds back into bug fixes and better CMake reference and release notes documentation.
+
+Another important source of CMake documentation is Kitware's official online ***CMake Reference Documentation***.
+This documentation is produced and versioned for every CMake release and contains a wealth of information including:
+
+* Reference material for the entire CMake language including for every supported command, property, variable and related topic
+* Documentation for the large collection of standard `*.cmake` modules that are installed along with CMake
+* Targeted guides including:
+  * CMake Tutorial (for developers contributing to projects that use CMake)
+  * User interaction Guide (for users trying to configure, build, install and test projects that use CMake)
+* Release notes for each version of CMake
+* Command-line documentation for the `cmake`, `ctest`, and `cpack` executables
+
+As a developer contributing to a project using CMake (i.e. editing the project's `CMakeLists.txt` files), when consulting the official reference documentation, it is advisable to only view documentation for the minimum version of CMake supported by the project (i.e. for the version `X.Y` listed in the project's top-level `CMakeLists.txt` file calling `cmake_minimum_required(VERSION X.Y)`).
+For example, if minimum version was 3.17, then one would select that version in the drop-down box (taking you to https://cmake.org/cmake/help/v3.17/).  This helps to avoid accidentally using CMake features not present in the minimum CMake version supported by the project.
+
+If one does not want to spend the $30 for the book "Professional CMake", then one may consider Kitware's ***Mastering CMake*** book.
+This is an updated, free online version of Kitware's classic "Mastering CMake" paperback book that used to be sold, (last updated for [CMake 3.1 published in 2015](https://www.amazon.com/Mastering-CMake-Ken-Martin/dp/1930934319)).
+It also contains tutorial coverage of key CMake concepts and other basic tutorial content (but usually at a less complete and detailed level compared to "Professional CMake").
+Note that the "guides" provided in the online CMake Reference Documentation are also exactly duplicated in the online "Mastering CMake" book.
+
+Finally, if one searches the above mentioned documentation sources "Professional CMake", the online CMake Reference Documentation, and "Mastering CMake" and does not find the answer, then one can turn to web searches.
```

Comment:
I would remove this sentence and the next two sentences.

I might add a note that when search on-line sources for CMake documentation, its a good idea to date-restrict the results to material less than one year old.

Reply:
As I mentioned [above](https://github.com/betterscientificsoftware/bssw.io/pull/1276#discussion_r831594517), you don't want to restrict your web searches to just recent hits but you also want to be careful to make sure this is the best advice given the current state of CMake.

The problem is that the internet contains a lot of material and advice on how to use CMake that is badly out of date given the current state of modern CMake.  But there are also long-standing problems with CMake with workarounds suggested by people 10+ years ago that are still valid today.  So relying on web searches is a double-edge sword.

How can I state this in a compact way?

Reply:
I shortened this some in https://github.com/betterscientificsoftware/bssw.io/pull/1276/commits/af525be727200d27e6db242b72b4c1fb1ffe61b3

Reply:
> So relying on web searches is a double-edge sword.

I think I see what you mean. But, I feel like this is true (to one extent or another) about almost *any* HPC software. I don't see what is *unique* about CMake in this regard that requires it being explicitly called out or explicitly alerting readers to the possiblity. Welcome to tech.

Reply:
That is true.  For example, advice on C++ given 10 years ago related to C++98 may not be the best advance with C++17.  So I guess it goes without saying as languages and tools evolve, what was good advice X years ago is not the best advice today.

So with that I can take out all mentions of web searches.

Reply:
Hopefully mostly addressed in commit https://github.com/betterscientificsoftware/bssw.io/pull/1276/commits/81408104691cb437122181e72f517b02feffe09a